Automattic/jetpack
# Jetpack SEO: Ensure we add post type support for custom fields

Fixes https://github.com/Automattic/jetpack/issues/38597

## Proposed changes:
* The issue was that it wasn't possible to add an SEO title or description for custom post types (they do not save).
* Adding the Jetpack SEO Meta fields to the Rest API for Custom post types proved challenging, however adding it directly using `add_post_type_support` does work. For this we also need to hook into `rest_api_init` as `init` is too early (before REST API is initialized).
* This also ensures that the 'Hide page from search engines' toggle works as well.
* As an addition, this PR also adds a filter that allows third party modification of the post types that will be processed to add support for custom fields if they do not already support it.
* Regarding the phan baseline addition - phan appears to be incorrectly linking the param name with the filtered param used in the function closure, so I've added it to the phan baseline file. The alternative would have been to create a named function with the logic, which adds unnecessary complexity.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/38597

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

**Self-hosted / Atomic**

To replicate the issue:

* On trunk, enable Testimonials: `/wp-admin/admin.php?page=jetpack#/writing`.
* Add a new testimonial. Try to add an SEO title and / or description, or using the 'Hide page from search engines' toggle. None will save.
* You can also visit `http://{yoursiteurl.com}/wp-json/wp/v2/jetpack-testimonial/{yourtestimonialid}` after saving, and notice there are no meta fields in the returned JSON.
* You can also visit the testimonial on the front-end and view source. The title / description you added should not be present.

To test the fix:
* Apply this PR.
* Follow the steps above - the fields / settings should save and be retained on page refresh. On visiting the REST API URL, notice the meta fields in the returned values. Also, the title / description you added should be visible in the source for the testimonial.

**Simple**

This feature is not available on Simple sites.
